providers/os/resources/purl/purl.go

```diff
@@ -43,44 +60,92 @@ func NewQualifiers(qualifier map[string]string) packageurl.Qualifiers {
 	return list
 }
 
-// NewPackageUrl creates a new package url for a given platform, name, version, arch, epoch and purlType
-// see https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst for more information
-func NewPackageUrl(pf *inventory.Platform, name string, version string, arch string, epoch string, purlType string) string {
-	qualifiers := map[string]string{}
-	if arch != "" {
-		qualifiers[QualifierArch] = arch
+// NewPackageURL creates a new package url for a given platform, name, version, and type.
+//
+// For more information, see:
+// https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst
+func NewPackageURL(pf *inventory.Platform, t Type, name, version string, modifiers ...Modifier) *PackageURL {
```

Comment:
That made me stumble.
First thought was, that we could derive the type from the platform. But no:
- Windows: windows, appx, nuget, ...
- Ubuntu: deb, snap, brew, ...
- ...

:face_with_diagonal_mouth:

Reply:
yeah, we really can't 😭